test/e2e/rollout/rollout_test.go

```diff
@@ -0,0 +1,313 @@
+			err := harness.CreateTestFleetWithSpec(fleetName, createFleetSpec(fleetName, bsqSuccess, lo.ToPtr(api.Percentage("100%"))))
+			Expect(err).ToNot(HaveOccurred())
+
+			deviceIDs := harness.StartMultipleVMAndEnroll(2)
+
+			labelsList := []map[string]string{
+				{"site": "madrid"},
+				{"site": "paris"},
+			}
+
+			err = harness.SetLabelsForDevicesByIndex(deviceIDs, labelsList, fleetName)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = applicationSpec.FromImageApplicationProviderSpec(applicationConfig)
+			Expect(err).ToNot(HaveOccurred())
+
+			deviceSpec := api.DeviceSpec{
+				Applications: &[]api.ApplicationProviderSpec{applicationSpec},
+			}
+
+			err = harness.CreateOrUpdateTestFleet(fleetName, testFleetSelector, deviceSpec)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Simulating a failure in the first batch")
+			err = SimulateDeviceFailure(harness)
+			Expect(err).ToNot(HaveOccurred())
+
+			time.Sleep(30 * time.Second)
+
+			By("Verifying that the rollout is paused due to unmet success threshold")
+			rolloutStatus, err := harness.GetRolloutStatus(fleetName)
+			Expect(err).ToNot(HaveOccurred())
+			//check rollout status tand reason to be api.ConditionStatusFalse and api.RolloutWaitingReason
+			Expect(rolloutStatus.Status).To(Equal(api.ConditionStatusFalse), "Rollout should be paused when success threshold is not met")
+			Expect(rolloutStatus.Reason).To(Equal(api.RolloutWaitingReason), "Rollout should be paused when success threshold is not met")
+
+			By("Fixing the failed device and verifying the rollout continues")
+			err = FixDeviceFailure(harness)
+			Expect(err).ToNot(HaveOccurred())
+
+			// Wait for rollout to continue
+			harness.WaitForBatchCompletion(fleetName, 2, 1*time.Minute)
+
+			// Verify that all devices are eventually updated
+			updatedDevices, err := harness.GetUpdatedDevices(fleetName)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(updatedDevices)).To(Equal(2), "All devices should be updated once threshold is met again")
+		})
+	})
+})
+
+func SimulateDeviceFailure(h *e2e.Harness) error {
+	// Define a command to simulate failure (e.g., stop a critical service or crash the VM)
+	failureCommand := []string{"sudo", "shutdown", "now"} // Example: Shut down the VM
+
+	// Run the failure command on the VM
+	_, err := h.VM().RunSSH(failureCommand, nil)
+	if err != nil {
+		return fmt.Errorf("failed to simulate failure on VM: %w", err)
+	}
+
+	return nil
+}
+
+func FixDeviceFailure(h *e2e.Harness) error {
+	// Define a command to fix the failure (e.g., restart the VM or start a critical service)
+	fixCommand := []string{"sudo", "reboot"} // Example: Reboot the VM
+
+	// Run the fix command on the VM
+	_, err := h.VM().RunSSH(fixCommand, nil)
+	if err != nil {
+		return fmt.Errorf("failed to fix failure on VM: %w", err)
+	}
+
+	// Wait for the VM to become ready again
+	err = h.VM().WaitForSSHToBeReady()
+	if err != nil {
+		return fmt.Errorf("VM %s did not recover after fix: %w", err)
```

Comment:
you probably already fixed this

```suggestion
		return fmt.Errorf("VM did not recover after fix: %w", err)
```